### PR TITLE
Discovery simplified [4739]

### PR DESCRIPTION
**Pull request by lemunozm**
#### Rationale: 
The use of the attribute `chosen` do not add a feature itself, it only gives to the user a concrete result for a concrete case. The same result can only be also obtained with few lines of code by the user, and with better management for his side. Furthermore, the "chosen" feature was forcing the use of reserve data for the ip into the `uxrAgentAddress` structure instead of a reference to an existant ip string.

#### Rationale:
There were two diferent function names: unicast and multicast. However, the behavior behind theses functions is not exactly what the name means. The unicast version can also receive several multicast ips, and the multicast version can only serach agent into the default multicast with the default port. For that, theses names have been changed.

- [ ] Updated API docs.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/142/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/103/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/84/)

File: include/uxr/client/profile/discovery/discovery.h [hunk 1]
```diff
@@ -29,18 +29,25 @@ extern "C"
...
     uint16_t port;
 
 } uxrAgentAddress;
 
-typedef bool (*uxrOnAgentFound) (const uxrAgentAddress* address, int64_t timestamp, void* args);
+typedef void (*uxrOnAgentFound) (const uxrAgentAddress* address, int64_t timestamp, void* args);
 
 
-UXRDLLAPI bool uxr_discovery_agents_multicast(uint32_t attempts, int period, uxrOnAgentFound on_agent_func, void* args, uxrAgentAddress* chosen);
+UXRDLLAPI void uxr_discovery_agents_default(uint32_t attempts,
```
Comment (julianbermudez) [inline include/uxr/client/profile/discovery/discovery.h]: Please use the code style guideline.

File: src/c/profile/discovery/discovery.c [hunk 2]
```diff
@@ -130,9 +130,8 @@ bool read_info_headers(ucdrBuffer* ub)
     return valid;
```
Comment (julianbermudez) [inline src/c/profile/discovery/discovery.c]: Maybe you could change the previous if by `return uxr_read_submessage_header(...);`

File: src/c/profile/discovery/discovery.c [hunk 3]
```diff
@@ -19,62 +19,65 @@
...
-    uxrAgentAddress* chosen;
     uxrOnAgentFound on_agent;
     void* args;
 
 } CallbackData;
 
 static void write_get_info_message(ucdrBuffer* ub);
-static bool listen_info_message(uxrUDPTransportDatagram* transport, int period, CallbackData* callback);
+static void listen_info_message(uxrUDPTransportDatagram* transport, int period, CallbackData* callback);
 static bool read_info_headers(ucdrBuffer* ub);
```
Comment (julianbermudez) [inline src/c/profile/discovery/discovery.c]: Why is it the only function that returns a boolean?

File: src/c/profile/discovery/discovery.c [hunk 4]
```diff
@@ -19,62 +19,65 @@
...
-static bool process_info(CallbackData* callback, Time_t timestamp, TransportLocator* transport);
+static void read_info_message(ucdrBuffer* ub, CallbackData* callback);
+static void process_info(CallbackData* callback, Time_t timestamp, TransportLocator* transport);
 
 //==================================================================
 //                             PUBLIC
 //==================================================================
 
-bool uxr_discovery_agents_multicast(uint32_t attemps, int period, uxrOnAgentFound on_agent_func, void* args, uxrAgentAddress* chosen)
+void uxr_discovery_agents_default(uint32_t attemps,
```
Comment (julianbermudez) [inline src/c/profile/discovery/discovery.c]: Please change `attemps` by `attempts`.

File: include/uxr/client/profile/discovery/discovery.h (hunk 1 above)
Comment (lemunozm) [reply include/uxr/client/profile/discovery/discovery.h]: Done!

File: src/c/profile/discovery/discovery.c (hunk 2 above)
Comment (lemunozm) [reply src/c/profile/discovery/discovery.c]: You are right. Done!

File: src/c/profile/discovery/discovery.c (hunk 3 above)
Comment (lemunozm) [reply src/c/profile/discovery/discovery.c]: Ok, I added a "boolean meaning" to the rest of the functions. In the future, we can interpret this value in order to improve some behaviour.

File: src/c/profile/discovery/discovery.c (hunk 4 above)
Comment (lemunozm) [reply src/c/profile/discovery/discovery.c]: Done!

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/147/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/108/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/89/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/163/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/123/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/104/)

**Pull request merged**